hbase-client/src/main/java/org/apache/hadoop/hbase/client/NormalizeTableFilterParams.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.client;
+
+import java.util.List;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+
+/**
+ * A collection of criteria used for table selection.
```

Comment:
What does it mean if these are all empty?

How are they handled if multiple are set?

Reply:
This defers to the existing functionality around listing table descriptors. I initially took a stab at documenting the behavior, but decided it didn't make sense to leave that attached to the PIOJO, dangling to rot. If you think otherwise, I can find that in my reflog and bring it back.

Reply:
I'll bring back that description.